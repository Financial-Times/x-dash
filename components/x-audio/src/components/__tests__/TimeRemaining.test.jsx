const { h } = require('@financial-times/x-engine');
const { shallow } = require('@financial-times/x-test-utils/enzyme');
const { TimeRemaining } = require('../TimeRemaining');

jest.mock('../classnames-helper', () => {
	return (...classes) => classes.join(' ');
});

describe('TimeRemaining', () => {

	const	props = {
		currentTime: 60*5,
		duration: 60*60 + 60*30 + 60*5
	};

	describe('given expanded is true', () => {
		test('should return text in HMMSS format', () => {
			const subject = shallow(<TimeRemaining {...props} expanded={true}/>);
			expect(subject).toMatchSnapshot();
		});
	})

	describe('given expanded is false', () => {

		test('should return text in "XX min remaining" format', () => {
			const subject = shallow(<TimeRemaining {...props} expanded={false}/>);
			expect(subject).toMatchSnapshot();
		});

		test('should return 1 if remaining minutes is less than 1', () => {
			const subject = shallow(<TimeRemaining {...props}
				expanded={false}
				currentTime={1}
				duration={60}
				/>);
			expect(subject).toMatchSnapshot();
		});
	});

});
