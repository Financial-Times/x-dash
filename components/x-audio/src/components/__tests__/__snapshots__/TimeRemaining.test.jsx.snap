// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeRemaining given expanded is false should not attempt to format non-numeric currentTime 1`] = `
<div
  className="audio-player__info__remaining"
/>
`;

exports[`TimeRemaining given expanded is false should not attempt to format non-numeric duration 1`] = `
<div
  className="audio-player__info__remaining"
/>
`;

exports[`TimeRemaining given expanded is false should return 1 if remaining minutes is less than 1 1`] = `
<div
  className="audio-player__info__remaining"
>
  1 min remaining
</div>
`;

exports[`TimeRemaining given expanded is false should return text in "XX min remaining" format 1`] = `
<div
  className="audio-player__info__remaining"
>
  90 min remaining
</div>
`;

exports[`TimeRemaining given expanded is true should display -00:00 if remaining seconds is less than 1 1`] = `
<div
  className="audio-player__info__remaining"
>
  -00:00
</div>
`;

exports[`TimeRemaining given expanded is true should not attempt to format non-numeric currentTime 1`] = `
<div
  className="audio-player__info__remaining"
/>
`;

exports[`TimeRemaining given expanded is true should not attempt to format non-numeric duration 1`] = `
<div
  className="audio-player__info__remaining"
/>
`;

exports[`TimeRemaining given expanded is true should return text in HMMSS format 1`] = `
<div
  className="audio-player__info__remaining"
>
  -1:30:00
</div>
`;
